core/src/main/java/org/ocpsoft/prettytime/PrettyTime.java

```diff
@@ -1514,35 +1515,50 @@ public PrettyTime setUnits(TimeFormat format, final TimeUnit... units)
     * Removes the mapping for the given {@link TimeUnit} type. This effectively de-registers the {@link TimeUnit} so it
     * will not be used in formatting. Returns the {@link TimeFormat} that was removed, or <code>null</code> if no unit
     * of the given type was registered.
+    *
+    * @deprecated Use {@link #removeUnit(ResourcesChronoUnit)} instead.
     */
+   @Deprecated
    public <UNIT extends TimeUnit> TimeFormat removeUnit(final Class<UNIT> unitType)
    {
       if (unitType == null)
          return null;
 
-      for (TimeUnit unit : units.keySet()) {
-         if (unitType.isAssignableFrom(unit.getClass())) {
-            cachedUnits = null;
+      for (ResourcesChronoUnit unit : units.keySet()) {
+         final TimeUnit timeUnit = unit.toTimeUnit();
+         if (unitType.isAssignableFrom(timeUnit.getClass())) {
+            cachedUnits.clear();
 
             return units.remove(unit);
          }
       }
       return null;
    }
 
+   /**
+    * Removes the mapping for the given {@link ResourcesChronoUnit} type. This effectively de-registers the
+    * {@link ResourcesChronoUnit} so it will not be used in formatting. Returns the {@link TimeFormat}
+    * that was removed, or <code>null</code> if no unit of the given type was registered.
+    */
+   public TimeFormat removeUnit(final ResourcesChronoUnit unit) {
+      if (unit == null) {
+         return null;
+      }
+
+      return units.remove(unit);
+   }
+
    /**
     * Removes the mapping for the given {@link TimeUnit}. This effectively de-registers the {@link TimeUnit} so it will
     * not be used in formatting. Returns the {@link TimeFormat} that was removed, or null if no such unit was
     * registered.
+    *
+    * @deprecated Use {@link #removeUnit(ResourcesChronoUnit)} instead.
     */
+   @Deprecated
    public TimeFormat removeUnit(final TimeUnit unit)
    {
-      if (unit == null)
```

Comment:
Please restore this method to its original state. `public TimeFormat removeUnit(final ResourcesChronoUnit unit)` should call this method.